### PR TITLE
[GITHUB-8] Add NPM Installation

### PR DESCRIPTION
**Pull request by dennisconrad**
Adds the functionality requested in #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,33 @@ Simply install latest LTS version.
         version: 4
 ```
 
+Install latest LTS version, latest lorem-ipsum and v0.15.3 of forever NPMs.
```
Comment (lobsterdore) [inline README.md]: This line is a bit confusing!

File: tasks/build.yml [hunk 2]
```diff
@@ -56,3 +56,18 @@
...
+    global: "{{ item.global | default('no' if item.path is defined else 'yes') }}"
+    ignore_scripts: "{{ item.ignore_scripts | default('no') }}"
+    name: "{{ item.name }}"
+    path: "{{ item.path | default('') }}"
+    production: "{{ item.production | default('no') }}"
+    registry: "{{ item.registry | default('') }}"
+    state: "{{ item.state | default('present') }}"
+    version: "{{ item.version | default('') }}"
+  with_items: "{{ nodejs.npms }}"
+  when: nodejs.npms
```
Comment (lobsterdore) [inline tasks/build.yml]: The preference is to define all possible variables in defaults.yml so it's easy to see all available variables for any role, with this is in mind can you define npms in defaults.yml like so:

```YAML
---

nodejs:
  npms: [ ]
  # 4 or 5
  version: 4
  workspace_user: ~
```
After this change you will not need the when clause, if nodejs.npms is empty Ansible will skip the task automatically.

Comment (dennisconrad) [reply tasks/build.yml]: Thank you!  I originally did not have all those `| default()` clauses, passing all variables through instead, but:

> [DEPRECATION WARNING]: Using variables for task params is unsafe, especially if
 the variables come from an external source like facts. This feature will be
removed in version 2.6.

I have changed the code as per your comment.

**Comment by lobsterdore** [comment]
LGTM 👍

**Pull request merged**